### DISCUSSION TITLE
Unable to use different accounts for github copilot chat and coding at the same time.

### DISCUSSION THREAD

**Discussion by Constantine-Ketskalo** [community/community#113587] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Seems like this [question](https://github.com/orgs/community/discussions/113156) is similar to my issue, but not identical and it's created as a question while I think mine is a bug.
I have 2 accounts on github: personal and working account. I've bought a github copilot from my personal account but some code that I'm working on I need to commit from the working account. Github copilot itself is authenticated through my personal account once, then I've switched visual studio settings to authenticate through my work account and I can use both github copilot via my personal account and github repository functionality via second account at the same time. But unfortunately seems like both pushing changes to repo and github copilot chat are directly referencing account that I'm currently logged with to visual studio itself. And it supports only 1 github account at a time. So whenever I use git repository functionality and I want to use github copilot chat or vise versa, I have to switch between the two github accounts in the visual studio account settings every time.
This is very inconvenient. Could you please fix it and make it possible to just login to both github and github copilot with account I bought it from and then use whatever other account I need for each individual repository?
Thanks.
I'm using Windows 11, Visual Studio Community 2022 (64-bit) Version 17.8.2.

**Comment by ayim** [comment]
@Constantine-Ketskalo we have some colleagues thinking about the multiple GitHub account space.

I wonder if one workaround is to login to GitHub Copilot completions extension (Chat does not support this) using machine auth.... you can attempt this using Tools > Options > GitHub Copilot > Sign-in with device flow. This will allow you to use Copilot Completions with an account that is NOT the account you've auth'd to GitHub with in Visual Studio.

![image](https://github.com/community/community/assets/10951869/c317499a-784c-438a-b497-839bc2eac728)


To make sure that your feedback is heard, I recommend you use the "Report a Problem" feature in Visual Studio.

![image](https://github.com/community/community/assets/10951869/7dbc143e-ead7-4ba8-8e9e-04bba89fb620)


@rheapatel @rub8n for awareness.

**Comment by terrylin104** [comment]
<img width="589" height="357" alt="{00570238-E537-4E45-9EDC-C3ED3B2F4280}" src="https://github.com/user-attachments/assets/7a5f1b95-019a-464b-8bf7-e4ecb6fd2381" />
Does it work?

**Reply by sticl** [reply]
this isn't visual studio right? it seems to be VS Code...

**Comment by Constantine-Ketskalo** [comment]
It's been couple years since that question already. And right now I think I was able to login to different accounts. Microsoft must have changed something in their code. I'm currently working on a totally different environment, but right now visual studio and github copilot are used from different accounts.
So either it was some weird local bug, or the code of visual studio and/or github copilot was updated.
